ui/src/components/collapsiblePanels/HorizontalPanel.tsx

```diff
@@ -42,20 +42,29 @@ const HorizontalPanel: React.FC<HorizontalPanelProps> = ({
 
   const classes = [
     baseClasses,
-    isOpen ? maxHeight ?? "h-64" : minHeight ?? "h-[36px]",
+    isOpen ? maxHeight ?? "h-100" : minHeight ?? "h-[36px]",
     className,
   ].reduce((acc, cur) => (cur ? `${acc} ${cur}` : acc));
 
   const handleToggle = useCallback(() => onToggle?.(!isOpen), [isOpen, onToggle]);
 
   const handleSelection = useCallback(
     (content: PanelContent) => {
+      // Close panel if already open
+      if (content.id === selected?.id && isOpen) {
+        handleToggle();
+        return;
+      }
+
+      // Switch panels
       setSelected(content);
+
+      // Open if closed
       if (!isOpen) {
         handleToggle();
       }
     },
-    [isOpen, handleToggle],
+    [isOpen, handleToggle, selected],
```

Comment:
Something like this is maybe clearer (this way I think the if statements are done in a way that the code speaks for itself and comments aren't needed)
```
if (content.id !== selected?.id) {
  setSelected(content);
  if (!isOpen) {
    onToggle?.(true)
  }
} else {
  onToggle?.(!isOpen)
}
```
and then remove the handleToggle since its not really necessary in my mind.

Reply:
Updated the function code. Thanks